packages/css/field.css

```
@@ -89,8 +91,19 @@
 
   /* Using double selector to ensure we win specificity */
   .ds-input.ds-input {
-    border-radius: 0;
     border: var(--dsc-field-affix-border);
     flex: 1 1 auto;
   }
+
+  /* if it has affix after */
+  .ds-input:has(+ [data-affix]) {
+    border-top-right-radius: 0;
+    border-bottom-right-radius: 0;
+  }
+
+  /* if it has affix before */
+  & [data-affix] + .ds-input {
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
+  }
```

Just came to think about it; everything inside `AffixWrapper` is either affix or input, so maybe we instead could do:
```suggestion
  /* if it has affix after */
  & .ds-input:not(:last-child) {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }

  /* if it has affix before */
  & .ds-input:not(:first-child) {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }
```
removing the need for `[data-affix]` which is nice in React, but even nicer when just writing plain HTML+CSS ☺️

---

It could in theory have anything in there, but yes, good point. 
It would break everything else if they put something else there 😓